src/hotspot/share/runtime/deoptimization.cpp

```diff
@@ -692,7 +687,9 @@ JRT_LEAF(BasicType, Deoptimization::unpack_frames(JavaThread* thread, int exec_m
   // It is actually ok to allocate handles in a leaf method. It causes no safepoints,
   // but makes the entry a little slower. There is however a little dance we have to
   // do in debug mode to get around the NoHandleMark code in the JRT_LEAF macro
-  ResetNoHandleMark rnhm; // No-op in release/product versions
+  // Beware though because allocating Handles must have a HandleMark or else the
+  // Handles will be leaked.
```

Comment:
This comment is confusing because there is HandleMark on L693.

Reply:
My point of this comment is that you should make sure you have a HandleMark if you've used ResetNoHandleMark in JRT_LEAF.